### PR TITLE
Scraper

### PR DESCRIPTION
**Pull request by ed3ath**

### CODE REVIEW

File: src/scrapers/market-scraper.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  }).catch((err) => console.log(`[${type.toLocaleUpperCase()} SCRAPER ERROR] ${err.message}`));
+}
+
+async function start(type, currentIndex = 0, endIndex = 9999, length = 3000) {
+  if (!process.env.HTTP_PROVIDER_URL) {
+    console.log('No HTTP_PROVIDER_URL, not scraping for market listings...');
+    return;
+  }
+  // eslint-disable-next-line no-nested-ternary
+  const nftAddress = (type === 'character' ? marketplaceHelper.getCharactersAddress() : (type === 'weapon' ? marketplaceHelper.getWeaponsAddress() : marketplaceHelper.getShieldsAddress()));
```
Comment (seiyria) [inline src/scrapers/market-scraper.js:29]: this code isn't very maintainable

it should have a let nftAddress; then a switch to check/set

File: src/scrapers/market-scraper.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    try {
+      console.log(`[${type.toLocaleUpperCase()} SCRAPER] Scraping ${length} from index ${currentIndex}`);
+
+      // eslint-disable-next-line no-await-in-loop
+      const list = await nftMarketPlace.methods.getListingSlice(nftAddress, currentIndex, length).call();
+
+      if (list.ids.length === 0) break;
+
+      for (let i = 0; i < list.ids.length; i++) {
+        // eslint-disable-next-line no-await-in-loop
```
Comment (seiyria) [inline src/scrapers/market-scraper.js:42]: you don't need this rule if you use for(let of..) IIRC

File: src/server.js [hunk 3]
```diff
@@ -129,4 +136,7 @@ DB.isReady.then(async () => {
   startTasks();
   startListeners();
   startLogging();
+  startScrapers('character');
```
Comment (seiyria) [inline src/server.js:139]: semantically, this function should start them all, but that's a minor nit

File: src/helpers/marketplace-helper.js [hunk 4]
```diff
@@ -120,6 +120,15 @@ const helpers = {
     return helpers.nftMarketPlace;
   },
 
+  getNftMarketPlace2: () => {
```
Comment (seiyria) [inline src/helpers/marketplace-helper.js:123]: this should get a different name, not a 2 on the end

File: src/helpers/marketplace-helper.js [hunk 5]
```diff
@@ -120,6 +120,15 @@ const helpers = {
...
+    const web3 = new Web3(process.env.HTTP_PROVIDER_URL);
```
Comment (seiyria) [inline src/helpers/marketplace-helper.js:124]: can you provide a sample fake url of what the http url should be? `https://x.xxx/<apikey>/xxx`

File: src/scrapers/market-scraper.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    console.log(`[${type.toLocaleUpperCase()} SCRAPER] Add ${marketplaceHelper.getTypeName(nftAddress)} ${nftId} from ${seller} for ${marketplaceHelper.realPrice(price)}`);
+  }).catch((err) => console.log(`[${type.toLocaleUpperCase()} SCRAPER ERROR] ${err.message}`));
+}
+
+async function start(type, currentIndex = 0, endIndex = 9999, length = 3000) {
+  if (!process.env.HTTP_PROVIDER_URL) {
+    console.log('No HTTP_PROVIDER_URL, not scraping for market listings...');
+    return;
+  }
+  // eslint-disable-next-line no-nested-ternary
```
Comment (seiyria) [inline src/scrapers/market-scraper.js:28]: don't add inline disables

File: .eslintrc.json [hunk 7]
```diff
@@ -16,6 +16,8 @@
       "global-require": "off",
       "max-len": ["error", 200],
       "linebreak-style": "off",
-      "import/no-dynamic-require": "off"
+      "import/no-dynamic-require": "off",
+      "no-plusplus": "off",
+      "no-param-reassign": "off"
```
Comment (seiyria) [inline .eslintrc.json:21]: don't change eslint rules

**Review by seiyria — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 7 above)
Comment (ed3ath) [reply .eslintrc.json:21]: image
i get lint errors

File: src/scrapers/market-scraper.js (hunk 2 above)
Comment (ed3ath) [reply src/scrapers/market-scraper.js]: if I use for(let of..), i will have to assign a different counter index for sellers and prices array

**Comment by seiyria** [comment]
thanks, but as you said I merged OPPandas PR. you should still collect a bounty from this since you put up a fair amount of work!

**Pull request closed without merging**